src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,485 @@
+                string bodyTitle = $@"The following errors have been detected when attempting to update dependencies in
+                    '{updateHistoryError.Repository}'
+                     {Environment.NewLine}";
+                issueUpdate.Body = $@"{bodyTitle}  {description}
+                    **/FyiHandle :** {_options.FyiHandle}";
+                _logger.LogInformation($"Updating issue body for the issue :  '{issue.Number}'");
+                await client.Issue.Update(repoId, issue.Number, issueUpdate);
+                return;
+            }
+            var comments = await client.Issue.Comment.GetAllForIssue(repoId, issue.Number);
+            // If issue body is replaced with newer description, then do not create a new comment with the same description.
+            foreach (var comment in comments)
+            {
+                if (shouldReplaceDescription(updateHistoryError.Arguments, comment.Body))
+                {
+                    // Issue's comment body will be updated
+                    _logger.LogInformation($"Updating comment body for the issue : '{issue.Number}'");
+                    await client.Issue.Comment.Update(repoId, comment.Id, description);
+                    return;
+                }
+            }
+            // New error for the same repo-branch combination, so adding a new comment to it. 
+            if (!string.IsNullOrEmpty(description))
+            {
+                _logger.LogInformation($"Creating a new comment for the issue : '{issue.Number}'");
+                await client.Issue.Comment.Create(repoId, issue.Number, description);
+            }
+        }
+
+        /// <summary>
+        /// Builds description and creates a func to check if the existing description has to be replaced or not.
+        /// </summary>
+        /// <param name="updateHistoryError">Error info for which github issue has to be created</param>
+        /// <param name="issueRepo">Repository where issue has to be created.</param>
+        /// <returns></returns>
+        private async Task IssueDescriptionEvaluator(RepositoryBranchUpdateHistoryEntry updateHistoryError, string issueRepo)
+        {
+            var description = "";
+            Func<string, string, bool> shouldReplaceDescription;
+            switch (updateHistoryError.Method)
+            {
+                // Only UpdateAssetAsync method has subscriptionId as one of the parameters in the arguments column. Other methods do not have this info.
+                case "UpdateAssetsAsync":
+                    description = ParseUpdateAssetsMethod(updateHistoryError);
+                    if (!string.IsNullOrEmpty(description))
+                    {
+                        shouldReplaceDescription = (argument, issueBody) =>
+                        {
+                            if (!string.IsNullOrEmpty(issueBody))
+                            {
+                                var argumentParser =  ParseIssueCommentBody(issueBody);
```

Comment:
I'm not entirely sure what this is, because it's "var" but it doesn't sound like it's a "parser", but rather just a collection of values.

Reply:
I parse issue/ comment body in the ParseIssueCommentBody method. From the marker comment I extract the subscriptionId, method and errorMessage. 

Perhaps I can rename this better - markerComponents ?

Reply:
That sounds like a better name. "argumentParser" makes it sound like something that you use to parse arguments... I'd expect it to have a "Parse" method on it, for example.